tms/__main__.py

```diff
@@ -119,24 +49,18 @@ async def main() -> None:
     # htcondor.enable_log()
     htcondor.enable_debug()
 
+    tmonitors: AppendOnlyList[TaskforceMonitor] = AppendOnlyList()
+
     loops = {
-        k: _create_loop_task(k)
-        for k in [
-            "starter",
-            "watcher",
-            "stopper",
-        ]
+        asyncio.create_task(scalar_loop(tmonitors)): "scalar",
+        asyncio.create_task(watcher_loop(tmonitors)): "watcher",
     }
-
-    while True:
-        done, _ = await asyncio.wait(
-            loops.values(),
-            return_when=asyncio.FIRST_COMPLETED,
-        )
-        # restart any done tasks
-        for key in loops:
-            if loops[key] in done:
-                loops[key] = _create_loop_task(key)
+    done, _ = await asyncio.wait(
+        loops.keys(),
+        return_when=asyncio.FIRST_COMPLETED,
+    )
+    for task in done:
+        LOGGER.error(f"'{loops[task]}' asyncio task completed: {task}")
```

Comment:
Not sure if it's better to use a `gather()` here?  Or `TaskGroup` if you want to be fancy.

We might want to cancel any remaining tasks once one of them has exited.  That at least eliminates the warning message you can get from python.

Reply:
`TaskGroup` is neat, trying it out now